crates/bevy_ecs/src/schedule_v3/condition.rs

```diff
@@ -0,0 +1,94 @@
+pub use helper::*;
+
+use crate::system::BoxedSystem;
+
+pub type BoxedCondition = BoxedSystem<(), bool>;
+
+/// Functions and closures that convert into [`System<In=(), Out=bool>`](crate::system::System)
+/// trait objects and have [read-only](crate::system::ReadOnlySystemParamFetch) parameters.
+pub trait Condition<Params>: sealed::Condition<Params> {}
+
+impl<Params, F> Condition<Params> for F where F: sealed::Condition<Params> {}
+
+mod sealed {
+    use crate::system::{
+        IntoSystem, IsFunctionSystem, ReadOnlySystemParamFetch, SystemParam, SystemParamFunction,
+    };
+
+    pub trait Condition<Params>: IntoSystem<(), bool, Params> {}
+
+    impl<Params, Marker, F> Condition<(IsFunctionSystem, Params, Marker)> for F
+    where
+        F: SystemParamFunction<(), bool, Params, Marker> + Send + Sync + 'static,
+        Params: SystemParam + 'static,
+        Params::Fetch: ReadOnlySystemParamFetch,
+        Marker: 'static,
+    {
+    }
+}
+
+pub mod helper {
```

Comment:
Another suggestion: event reader of type `E` has at least one entry.

Reply:
That is tricky. 

If the test is "event reader has at least one entry *that has not been seen*" (which is what I think you mean) then the condition system would advance its event reader if `true` or not if `false`. The problem is if *other* conditions return `false`, we can't reset the event reader. `last_change_tick` is the only system state we can tamper with right now.

If the condition doesn't advance its event reader, it'd just return `true` unless the global event queue was actually empty.

Reply:
🤔 Right, I see your point. We should scrap this idea then.

Reply:
> The problem is if other conditions return false, we can't reset the event reader. 

I would expect this behavior. `iyes_loopless` works this way.

Reply:
Not resetting the reader would be completely wrong. 

All changes that have occurred "after the system (set) last ran" should be detected. If something's other conditions fail, then it doesn't run, and so advancing the reader will arbitrarily miss events.

Reply:
I think it depends on the use case. For example, it would be perfectly fine for me to ignore an event when another condition is not met.

Reply:
I'd say just write your own condition then. 

An `event_pending` helper that behaves differently when paired with other conditions is not good (especially considering change-filtered queries don't have the same issue).

Reply:
Yep agreed: these will be easy to write for users.